language/e2e-testsuite/src/tests/verify_txn.rs

```diff
@@ -29,334 +29,339 @@ use vm::file_format::CompiledModule;
 
 #[test]
 fn verify_signature() {
-    let mut executor = FakeExecutor::from_genesis_file();
-    executor.set_golden_file(current_function_name!());
-    let sender = executor.create_raw_account_data(900_000, 10);
-    executor.add_account_data(&sender);
-    // Generate a new key pair to try and sign things with.
-    let private_key = Ed25519PrivateKey::generate_for_testing();
-    let program = encode_peer_to_peer_with_metadata_script(
-        account_config::xus_tag(),
-        *sender.address(),
-        100,
-        vec![],
-        vec![],
-    );
-    let signed_txn = transaction_test_helpers::get_test_unchecked_txn(
-        *sender.address(),
-        0,
-        &private_key,
-        sender.account().pubkey.clone(),
-        Some(program),
-    );
-
-    assert_prologue_parity!(
-        executor.verify_transaction(signed_txn.clone()).status(),
-        executor.execute_transaction(signed_txn).status(),
-        StatusCode::INVALID_SIGNATURE
-    );
+    test_with_different_versions! {[1, 2], |mut test_env| {
```

Comment:
Why did we test the different versions in the `verify_txn` only? Was wondering if it would make more sense to test transactions other than those that would trigger verification error? Because for this release, we want to make sure the shim layer is translating the txn execution logic into script functions correctly. Testing transactions that would be disacarded before execution phase doesn't seem quite helpful there.

Reply:
Agreed! These are just example ones so we can see what the translation will look like. I think we also want to check to make sure script functions are allowed through/rejected depending on the version number as well I think.